kong/llm/drivers/openai.lua

```diff
@@ -11,8 +11,28 @@ local socket_url = require "socket.url"
 local DRIVER_NAME = "openai"
 --
 
+-- merge_defaults takes the model options, and sets any defaults defined,
+-- if the caller hasn't explicitly set them
+--
+-- we have already checked that "max_tokens" isn't overridden when it
+-- is not allowed to do so.
+local function merge_defaults(request, options)
+  local merging = {
+    [1] = "max_tokens",
+    [2] = "temperature",
+    [3] = "top_p",
+    [4] = "top_k",
+  }
+
+  for i, v in ipairs(merging) do
+    request[v] = request[v] or (options and options[v]) or nil
+  end
+
+  return request
+end
```

Comment:
IDK if luajit is smart enough to identify that this table is const and avoid re-creating it, so I think I'd recommend moving its creation to the outer scope.

Also, caching an empty table and swapping it in saves us some operations in the loop.

```suggestion
local merge_defaults
do
  local merging = {
    [1] = "max_tokens",
    [2] = "temperature",
    [3] = "top_p",
    [4] = "top_k",
  }

  local empty = {}

  function merge_defaults(request, options)
    options = options or empty
    
    for i, v in ipairs(merging) do
      request[v] = request[v] or options[v] or nil
    end

    return request
  end
end
```